File: DsDotNet/src/Engine/Engine.Parser/0.ParserGlobal.cs
global using System;
global using System.Linq;
global using System.Collections.Generic;
global using System.Diagnostics;
global using static System.Diagnostics.Debug;
//global using System.Reactive.Linq;
global using log4net;
global using Antlr4.Runtime;
global using Antlr4.Runtime.Tree;
global using Engine.Core;

global using static Engine.Parser.dsParser;
global using static Engine.Parser.DsParser;
global using static Engine.Core.CoreModule;
global using static Engine.Core.Interface;

namespace Engine.Parser;
public static class Global
{
    public static ILog Logger { get; set; }
}

File: DsDotNet/src/Engine/Engine.Parser.FS/0.ParserGlobal.fs
namespace Engine.Parser.FS

open System
open System.Linq
open System.Collections.Generic
open System.Diagnostics
open type System.Diagnostics.Debug
//open System.Reactive.Linq
open log4net
open Antlr4.Runtime
open Antlr4.Runtime.Tree
open Antlr4.Runtime.Misc
open Engine.Common
open Engine.Core

open type Engine.Parser.dsParser
open type Engine.Parser.DsParser
//open type Engine.Core.CoreModule
//open type Engine.Core.Interface
//open type Engine.Core.SpitModuleHelper
open type Engine.Parser.Global

//type Global = {
//    Logger:ILog

//    /// <summary> Do nothing </summary>
//    public static void NoOp() { }

//    public static void LogDebug(object message) => Logger?.Debug(message)
//    public static void LogInfo(object message) => Logger?.Info(message)
//    public static void LogWarn(object message) => Logger?.Warn(message)
//    public static void LogError(object message) => Logger?.Error(message)

//    public static void Verify(string message, bool condition)
//    {
//        if (!condition)
//        {
//            LogError(message)
//            throw new Exception(message)
//        }
//    }
//}

